opsdroid/events.py

```diff
@@ -0,0 +1,166 @@
+"""Classes to describe different kinds of possible event."""
+
+from abc import ABC
+from datetime import datetime
+from copy import copy
+import asyncio
+from random import randrange
+
+from opsdroid.helper import get_opsdroid
+
+
+# pylint: disable=too-few-public-methods
+class Event(ABC):
+    """A generic event type.
+
+    Initiates an Event object with the most basic information about its
+    creation.
+
+    Args:
+        user: String name of user sending event
+        room: String name of the room or chat channel in which event was sent
+        connector: Connector object used to interact with given chat service
+
+    Attributes:
+        created: Local date and time that message object was created
+        user: String name of user sending message
+        room: String name of the room or chat channel in which message was sent
+        connector: Connector object used to interact with given chat service
+        responded_to: Boolean initialized as False. True if message has been
+            responded to
+
+    """
+
+    def __init__(self, user, room, connector):  # noqa: D107
+        self.created = datetime.now()
+        self.user = user
+        self.room = room
+        self.connector = connector
+        self.responded_to = False
+
+
+class Message(Event):
+    """A message object.
+
+    Stores messages in a format that allows OpsDroid to respond or react with
+    delays for thinking and typing as defined in configuration YAML file.
+
+    Args:
+        user: String name of user sending message
+        room: String name of the room or chat channel in which message was sent
+        connector: Connector object used to interact with given chat service
+        text: String text of message
+        raw_message: Raw message as provided by chat service. None by default
+
+    Attributes:
+        created: Local date and time that message object was created
+        user: String name of user sending message
+        room: String name of the room or chat channel in which message was sent
+        connector: Connector object used to interact with given chat service
+        text: Text of message as string
+        raw_message: Raw message provided by chat service
+        regex: A re match object for the regular expression message was matched
+            against
+        responded_to: Boolean initialized as False. True if message has been
+            responded to
+
+    """
+
+    def __init__(self, user, room, connector,
+                 text, raw_message=None):  # noqa: D107
```

Comment:
I think it might be worth moving the `raw_message` kwarg up into `Event`. Certainly from the matrix perspective the raw event would be useful for much more than just messages. I would imagine most connectors will have a json or similar event structure that might be worth carrying around?

Reply:
Maybe it should be `raw_event`?

Reply:
That's what I've gone with, yes.